openverse_catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -8,266 +8,172 @@
+
+    def get_license_url(self, license_field) -> str | None:
+        if len(license_field) > 1:
+            logger.warning("More than one license field found")
+        for license_ in license_field:
+            if "creativecommons" in license_:
+                return license_
+        return None
+
+    def get_foreign_landing_url(self, data: dict) -> str:
+        original_url = data.get("edmIsShownAt")
+        if original_url is not None:
+            return original_url[0]
+        europeana_url = data.get("guid")
+        return europeana_url
+
+    def get_meta_data_dict(self, data: dict) -> dict:
+        meta_data = {
+            "country": data.get("country"),
+            "dataProvider": data.get("dataProvider"),
+            "description": self.get_description(data),
+        }
+
+        return {k: v for k, v in meta_data.items() if v is not None}
+
+    def get_description(self, data: dict) -> str | None:
+        lang_aware_description = data.get("dcDescriptionLangAware")
+        if lang_aware_description:
+            description = lang_aware_description.get(
+                "en"
+            ) or lang_aware_description.get("def")
+        else:
+            description = data.get("dcDescription")
 
-DELAY = 30.0
-RESOURCES_PER_REQUEST = "100"
-PROVIDER = prov.EUROPEANA_DEFAULT_PROVIDER
-API_KEY = Variable.get("API_KEY_EUROPEANA", default_var=None)
-ENDPOINT = "https://www.europeana.eu/api/v2/search.json?"
-# SUB_PROVIDERS is a collection of providers within europeana which are
-# valuable to a broad audience
-SUB_PROVIDERS = prov.EUROPEANA_SUB_PROVIDERS
-
-RESOURCE_TYPE = "IMAGE"
-REUSE_TERMS = ["open", "restricted"]
-
-DEFAULT_QUERY_PARAMS = {
-    "profile": "rich",
-    "reusability": REUSE_TERMS,
-    "sort": ["europeana_id+desc", "timestamp_created+desc"],
-    "rows": RESOURCES_PER_REQUEST,
-    "media": "true",
-    "start": 1,
-    "qf": [f"TYPE:{RESOURCE_TYPE}", "provider_aggregation_edm_isShownBy:*"],
-}
-
-delayed_requester = DelayedRequester(DELAY)
-image_store = ImageStore(provider=PROVIDER)
-
-
-def main(date):
-    logger.info(f"Processing Europeana API for date: {date}")
-
-    start_timestamp, end_timestamp = _derive_timestamp_pair(date)
-    _get_pagewise(start_timestamp, end_timestamp)
-
-    total_images = image_store.commit()
-    logger.info(f"Total images: {total_images}")
-    logger.info("Terminated!")
-
-
-def _get_pagewise(start_timestamp, end_timestamp):
-    cursor = "*"
+        if description:
+            return description[0].strip()
 
-    while cursor is not None:
-        image_list, next_cursor, total_number_of_images = _get_image_list(
-            start_timestamp, end_timestamp, cursor
-        )
+        return ""
 
-        if next_cursor is None:
-            break
 
-        cursor = next_cursor
+class EuropeanaDataIngester(ProviderDataIngester):
+    providers = {"image": prov.EUROPEANA_DEFAULT_PROVIDER}
+    sub_providers = prov.EUROPEANA_SUB_PROVIDERS
+    batch_limit = 100
```

Comment:
I believe this is the default batch limit, so it shouldn't need to be overwritten here!